### PR TITLE
Add EFA and SRD support for ib_send_* tests

### PR DESCRIPTION
**Pull request by firasj**
Add support for Scalable Reliable Datagram (SRD) connection type which runs
over Amazon Elastic Fabric Adapter (EFA).

### CODE REVIEW

File: src/perftest_parameters.c [hunk 1]
```diff
@@ -807,8 +807,15 @@ static void change_conn_type(int *cptr, VerbType verb, const char *optarg)
 		fprintf(stderr," DC not detected in libibverbs\n");
 		exit(1);
 		#endif
+	} else if (strcmp(connStr[6], optarg) == 0) {
+		#ifdef HAVE_SRD
```
Comment (dimasique1) [inline src/perftest_parameters.c]: as far as I understand, SRD works for SEND only. If so, add check for verb as it is done for UC and UD

File: src/send_bw.c [hunk 2]
```diff
@@ -378,9 +378,8 @@ int main(int argc, char *argv[])
 	}
 
 	if (user_param.test_method == RUN_ALL) {
-
-		if (user_param.connection_type == UD)
-			size_max_pow =  (int)UD_MSG_2_EXP(MTU_SIZE(user_param.curr_mtu)) + 1;
+		size_max_pow = MSG_SZ_2_EXP(user_param.size) + 1;
```
Comment (dimasique1) [inline src/send_bw.c]: Why is it needed?

File: src/perftest_parameters.c (hunk 1 above)
Comment (firasj) [reply src/perftest_parameters.c]: Right. Added check.

File: src/send_bw.c (hunk 2 above)
Comment (firasj) [reply src/send_bw.c]: Previously, the BW tests with RUN_ALL ran all sizes 2^(1...24).
SRD does not support sending messages this big, and it is my understanding that other connections won't be able to send messages if the size is greater than what's reported in max message size in port attributes.

Comment (dimasique1) [reply src/send_bw.c]: Let's not change the behavior and put a condition here for SRD

Comment (firasj) [reply src/send_bw.c]: Ack.

**Comment by firasj** [comment]
Modified PR to have minimal changes on other connection types.

**Comment by firasj** [comment]
There might've been a small issue with my latest revision.
Please hold off merging for now.
Thanks

**Comment by firasj** [comment]
Hi, I have updated the PR and it's ready for review/submission.

Thanks,
Firas

File: src/perftest_resources.c [hunk 3]
```diff
@@ -1047,9 +1052,10 @@ int destroy_ctx(struct pingpong_context *ctx,
...
 	for (i = first; i < user_param->num_of_qps; i++) {
 
-		if (( (user_param->connection_type == DC && !((!(user_param->duplex || user_param->tst == LAT) && (user_param->machine == SERVER) )
-							|| ((user_param->duplex || user_param->tst == LAT) && (i >= num_of_qps)))) ||
-					user_param->connection_type == UD) && (user_param->tst == LAT || user_param->machine == CLIENT || user_param->duplex)) {
+		if (((user_param->connection_type == DC && !((!(user_param->duplex || user_param->tst == LAT) && user_param->machine == SERVER)
+							|| ((user_param->duplex || user_param->tst == LAT) && i >= num_of_qps))) ||
+					user_param->connection_type == UD || user_param->connection_type == SRD) &&
+				(user_param->tst == LAT || user_param->machine == CLIENT || user_param->duplex)) {
 			if (ibv_destroy_ah(ctx->ah[i])) {
```
[the remark this answers is not in the document]
Comment (firasj) [reply src/perftest_resources.c]: We do not support rss.

**Comment by dimasique1** [comment]
okay, LGTM. Will test and merge later next week.

**Comment by firasj** [comment]
Updated PR to fix a compilation bug when using new WR API.

**Comment by galpress** [comment]
@dimasique1 Any updates?

**Comment by dimasique1** [comment]
Sorry for a long wait, will do my best to merge all PRs by the end of this week.

**Pull request merged**